x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/user/GetUserPrivilegesResponseTests.java

```diff
@@ -159,7 +212,36 @@ private GetUserPrivilegesResponse randomResponse() {
             )
         );
         final Set<String> runAs = randomStringSet(3);
-        return new GetUserPrivilegesResponse(cluster, conditionalCluster, index, application, runAs);
+        final Set<GetUserPrivilegesResponse.RemoteIndices> remoteIndex = allowRemoteIndices
+            ? Sets.newHashSet(
+                randomArray(
```

Comment:
Randomize choosing an empty set here? Or in callers to randomResponse() versus randomResponse(true)?

I am wondering if it makes sense to name the methods randomResponse() and randomResponseRcs2(), and add a helper randomResponseRcs1OrRcs2() when callers want to randomize between RCS 1.0 vs 2.0.

Reply:
> Randomize choosing an empty set here?

This is covered by the fact that we use `randomArray` which may be empty.

> I am wondering if it makes sense to name the methods randomResponse() and randomResponseRcs2(), and add a helper randomResponseRcs1OrRcs2() when callers want to randomize between RCS 1.0 vs 2.0.

Thanks for raising. I just realized that `randomResponse()` _should_ actually include remote indices makes the default behavior less confusing.  Now the suite uses `randomResponse()` everywhere, except one method where it's used as:

```
final Version version = VersionUtils.randomCompatibleVersion(random(), Version.CURRENT);
final boolean canIncludeRemoteIndices = version.onOrAfter(RoleDescriptor.VERSION_REMOTE_INDICES);
final GetUserPrivilegesResponse original = randomResponse(canIncludeRemoteIndices);
```  

Which is hopefully made less ambiguous by the var name. Let me know if you'd still prefer to have an explicit `randomResponseWithoutRemoteIndices` for the above test.

Reply:
I would prefer to have different names. I will leave it up to you to decide on the names.

I would caution putting "RemoteIndices" in the name, though. There are 5 QC privileges (indices, run_as, cluster, global, applications). So far we added remote_indices, but is there a possibility we will get asked to add analogous settings for the other 4 privileges. For example, remote_run_as, remote_cluster, remote_global, or remote_applications.

Reply:
Unrelated observation:
- The design copied `indices` to `remote_indices`, and added a list called `remote_indices.remote_clusters`.
- If we follow the same pattern for `cluster`, it ends up with a list called `remote_cluster.remote_clusters`.

Reply:
For now, I'll stick with the current method names. `randomResponse()` is self-explanatory. To avoid redundant code, we want a private method with a signature along the lines of `randomResponse(boolean allowRemoteIndices)`. Since this is a private helper method inside a single test suite, it's going to be straightforward to rename and tweak should we ever add other types of remote privileges (for which there currently is no medium term requirement). Should this requirement come up, happy to revisit this!